ledger/participant-integration-api/src/main/scala/platform/apiserver/error/ErrorInterceptor.scala

```diff
@@ -70,13 +72,13 @@ final class ErrorInterceptor extends ServerInterceptor {
           val newMetadata =
             Option(Status.trailersFromThrowable(selfServiceException)).getOrElse(new Metadata())
           val newStatus = Status.fromThrowable(selfServiceException)
-          super.close(newStatus, newMetadata)
+          LogOnUnhandledFailureInClose(delegate().close(newStatus, newMetadata))
```

Comment:
Switched to passing `delegate().close(..)` instead of `super.close(...)` by name to error handler otherwise it throws `IllegalAccessError` at runtime (e.g. in `ErrorInterceptorSpec`). I'm not fully happy about changing, but exhausted some options and am considering that it is clear to devs what's happening here. If someone has an idea, I'd appreciate it (CC @skisel @pbatko-da @nmarton-da )

Reply:
I think this is a Scala compiler bug unfortunately. When the `super.close` is wrapped in a closure, Scala creates an anonymous class around the dispatch site. That class, then, tries to actually invoke `super.close` which delegates to `PartialForwardingServerCall` which has default Java visibility (i.e. it is package-private). The anonymous Scala class does not have access rights to that class since it lives in a different package, which translates to an `IllegalAccessError`

The solution is to use a delegate private method in this class that separates the unrelated accessor from the non-public Java class.

Reply:
Validated and implemented workaround: 104ff1b7a5624762c341736ba05a6ddaf9781966